Marketing Information is fetched with ReleaseToggleCollectionGateway

vokoblin opened the pull request:
Adds some extra marketing information to be fetched.

Commit:
Marketing Information is fetched with ReleaseToggleCollectionGateway

Commit:
Adds toggle_marketing_information table into tables.sql

Commit:
Making each field of toggle marketing information optional

Commit:
Testing all the getters

Commit:
Merge branch 'master' of github.com:clearbooks/labs-mysql into LABS-67_MysqlRleaseToggleCollectionGateway

Conflicts:
	composer.lock
	test/Release/MysqlReleaseGatewayTest.php

Commit:
Refactoring to remove complexity problems

Commit:
Fixing return types

Commit:
Code formating

Commit:
Refactoring to fix complexity

Commit:
MORE Refactoring to fix complexity

Clearbooks-Tom commented on src/Toggle/MysqlGetAllTogglesGateway.php:
```diff
@@ -14,14 +14,123 @@
+     * @param $row
+     * @return string
+     */
+    protected function setDefaultForDescriptionOfToggle( $row )
```
I'd be inclined to rename this to get*****

Clearbooks-Tom replied:
where the asterisks denote the original method name sans "set"

Clearbooks-Tom commented on src/Toggle/MysqlGetAllTogglesGateway.php:
```diff
@@ -14,14 +14,123 @@
+            $row[ 'name' ],
+            $row[ 'release_id' ],
+            (bool) $row[ 'is_active' ],
+            $screenshotUrl,
```
so instead of saving all of the various values for the marketing information into variables first why not just call the relevant get***\* functions from here

MeliaBurnley replied:
I second this, would be slightly easier than a massive `list ( ... )` thing

Commit:
Making more sense with method naming

peter-horvath commented on src/Toggle/MysqlGetAllTogglesGateway.php:
```diff
@@ -13,15 +13,107 @@
+     */
+    protected function getDefaultForScreenshotUrl( $row )
+    {
+        $screenshotUrl = isset( $row[ 'screenshot_urls' ] ) ? $row[ 'screenshot_urls' ] : null;
```
In all these getDefault methods it's quite unnecessary to assign the value to a local variable, you could just return with the result right away.

peter-horvath commented on src/Toggle/MysqlGetAllTogglesGateway.php:
```diff
@@ -13,15 +13,107 @@
+     */
+    protected function getToggleFromRow( $row )
+    {
+        return new Toggle(
```
You don't have to use it if you don't want to, but Clearbooks\Labs\Db\Entity\CamelCaseMapperEntity would probably make things easier.

peter-horvath commented on test/Release/MysqlReleaseToggleCollectionGatewayTest.php:
```diff
@@ -43,6 +43,7 @@ public function setUp()
 
     public function tearDown()
     {
+        $this->deleteAddedTogglesMarketingInformation();
```
There's a LabsTest.php abstract class in labs-php-client-mysql. Probably you should use that in the feature, it uses a transaction and rolls it back at tearDown, so you wouldn't need to write code for undoing DB modifications.

Commit:
Does the refactoring

Commit:
Merge branch 'master' of github.com:clearbooks/labs-mysql into LABS-67_MysqlRleaseToggleCollectionGateway

Conflicts:
	src/Toggle/Entity/Toggle.php

Commit:
More Refactoring

Commit:
Updates composer

Commit:
Updating composer again

Commit:
Adds a Bootstrap php file and changes phpunit

Commit:
Updating travis before_script

Commit:
Updating travis again

Commit:
Update travis again and again

Commit:
Updating travis for the last time.

peter-horvath commented on db/1.sql:
```diff
@@ -55,5 +55,21 @@ CREATE TABLE `group_activated_toggle` (
+	INDEX `FK_toggle_marketing_information_toggle` (`toggle_id`),
+	CONSTRAINT `FK_toggle_marketing_information_toggle` FOREIGN KEY (`toggle_id`) REFERENCES `toggle` (`id`)
+)
+COLLATE='latin1_swedish_ci'
```
This should be utf8mb4_unicode_ci.

peter-horvath commented on test/bootstrap.php:
```diff
@@ -0,0 +1,14 @@
+use Clearbooks\Labs\Bootstrap;
+use Clearbooks\Labs\Db\DbDIDefinitionProvider;
+require_once __DIR__ . "/../vendor/autoload.php";
+require_once __DIR__ . "/../vendor/clearbooks/labs-php-client-mysql/src/Bootstrap.php";
```
Hm does this not work without this line? I think the vendor/autoload.php should make it available.

vokoblin replied:
I'll try, :+1:

Commit:
Makes changes:
changes COLLATE from latin1 to utf8mb4.
deletes un-used __DIR__."/../vendor..."

Commit:
Re-test
